app/services/generate_new_week_assignments.rb

```diff
@@ -24,16 +24,31 @@ def generate_last_assignment_list
 
     def last_employee_assignment(employee)
       Assignment.where(employee: employee).order(:week_id, :id).last ||
-        Assignment.new(week: Week.new(starts_at: employee.created_at, office: @current_office), employee: employee, created_at: employee.created_at)
+        Assignment.new(week: Week.new(starts_at: employee.created_at-1.week, office: @current_office), employee: employee, created_at: employee.created_at)
     end
 
     def create_week_assignment
-      if @last_assignment_list.count >= 1
-        employee = @last_assignment_list.first.employee
+      count = @last_assignment_list.count
+
+      if count >= 2
+        @new_assignment = Assignment.new
 
-        Assignment.create(week: @week, employee: employee)
+        while !@new_assignment.save 
+          arr_index = exponential_distribution(count)
 
-        @last_assignment_list.shift
+          employee = @last_assignment_list[arr_index].employee
+
+          @new_assignment = Assignment.new(week: @week, employee: employee)
+        end
+      elsif count == 1
+        employee = @last_assignment_list.first.employee
+
+        Assignment.create(week: @week, employee: employee) 
       end
     end
+
+    #returns a random integer between 0-max heavily weighted towards 0
+    def exponential_distribution(max)
+      ((-Math.log(rand)/0.9).ceil - 1) % max
```

Comment:
Neato! Did you look at the results of this? Does it create the desired effect of having people with occasionally different partners?